### PR TITLE
[prompty][Local] Load the .env file before test/batch run prompty

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
```
pf flow test --flow path/to/prompty/prompty --env   # load env from .env
pf flow test --flow path/to/prompty/prompty --env path/to/env.env
pf flow test --flow path/to/prompty/prompty --env KEY=VALUE
pf flow test --flow path/to/prompty/prompty --env path/to/env.env KEY=VALUE # with env override

pf run create --flow path/to/prompty/prompty --env   # load env from .env
pf run create --flow path/to/prompty/prompty --env path/to/env.env
pf run create --flow path/to/prompty/prompty --env KEY=VALUE
pf flow test --flow path/to/prompty/prompty --env path/to/env.env KEY=VALUE # with env override

```
Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_cli/_params.py [hunk 1]
```diff
@@ -94,8 +114,9 @@ def add_param_set_positional(parser):
 
 def add_param_environment_variables(parser):
     parser.add_argument(
+        "-env",
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_params.py:117]: we don't need to support -env , --env is ok

File: src/promptflow-devkit/promptflow/_cli/_params.py [hunk 2]
```diff
@@ -38,6 +41,23 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
             return super().get_action(values, option_string)
 
 
+class EnvironmentVariablesAction(AppendToDictAction):  # pylint: disable=protected-access
+    def get_action(self, values, option_string):  # pylint: disable=no-self-use
+        if len(values) == 1 and "=" not in values[0]:
+            # Load environment variables in .env file.
+            if values[0].endswith(".env"):
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_params.py:48]: not sure if we should support multiple .env files

File: src/promptflow-devkit/promptflow/_cli/_params.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow-devkit/promptflow/_cli/_params.py]: change to --env

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_cli/_params.py [hunk 3]
```diff
@@ -38,6 +41,23 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
             return super().get_action(values, option_string)
 
 
+class EnvironmentVariablesAction(AppendToDictAction):  # pylint: disable=protected-access
+    def get_action(self, values, option_string):  # pylint: disable=no-self-use
+        if len(values) == 1 and "=" not in values[0]:
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_params.py:46]: add load_dotenv support when len ==0

Comment (lalala123123) [reply src/promptflow-devkit/promptflow/_cli/_params.py]: fix it

File: src/promptflow-devkit/promptflow/_cli/_params.py [hunk 4]
```diff
@@ -95,11 +134,12 @@ def add_param_set_positional(parser):
 def add_param_environment_variables(parser):
     parser.add_argument(
         "--environment-variables",
-        action=AppendToDictAction,
+        "--env",
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_params.py:137]: can this be omitted?

Comment (lalala123123) [reply src/promptflow-devkit/promptflow/_cli/_params.py]: omit it

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**